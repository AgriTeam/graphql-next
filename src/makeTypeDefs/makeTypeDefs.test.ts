import 'jest';
import { GraphQLSchema, buildSchema } from 'graphql';
import { generateTsFromGql, getTypeDefs} from './';
import simpleSchema from './test/types';
import { schemaToInterfaces } from '@gql2ts/from-schema';
import * as prettyFormat from 'pretty-format';

describe('gql Types to Typescript types', () => {
    // TODO: should work with union tyoes
    it.skip('should return merged typedefs', async () => {
        const typeDefs = await getTypeDefs('**/*.gql');
        expect(prettyFormat(typeDefs)).toMatchSnapshot();
    });
    it.skip('merged types should be convertable to a GraphQLSchema', async () => {
        const typeDefs = await getTypeDefs('**/*.gql');
        expect(buildSchema(typeDefs)).toBeInstanceOf(GraphQLSchema);
    });
    it('merged types schema', async () => {
        // const namespaceOpts = { ignoreTypeNameDeclaration: true};
        const actual: string = schemaToInterfaces(simpleSchema, {
            ignoredTypes: []
          });
        expect(prettyFormat(actual)).toMatchSnapshot();
    });
    it.skip('End to End test: should create typescript types from graphql files', async () => {
        const tsTypesNameSpace = await generateTsFromGql({globPattern: '**/*.gql'});
        expect(prettyFormat(tsTypesNameSpace)).toMatchSnapshot();
    });
});
