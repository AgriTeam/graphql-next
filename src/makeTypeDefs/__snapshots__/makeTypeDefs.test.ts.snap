// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gql Types to Typescript types End to End test: should create typescript types from graphql files 1`] = `
"\\"// tslint:disable
// graphql typescript definitions

declare namespace DH {
  interface IGraphQLResponseRoot {
    data?: IQuery;
    errors?: Array<IGraphQLResponseError>;
  }

  interface IGraphQLResponseError {
    message: string;            // Required for all errors
    locations?: Array<IGraphQLResponseErrorLocation>;
    [propName: string]: any;    // 7.2.2 says 'GraphQL servers may provide additional entries to error'
  }

  interface IGraphQLResponseErrorLocation {
    line: number;
    column: number;
  }


  interface IQuery {
    dvptCooperation: Array<IBasicIndicator> | null;
    topTchRecipients: Array<ITopTchRecipients> | null;
    /**
    description: id is district slug
  */
    overviewTabRegional: IOverviewTabRegional | null;
  }


  interface IBasicIndicator {
    id: string;
    year: number;
    color: string;
    value: number;
    uid: string;
  }


  interface ITopTchRecipients {
    recipient: string;
    value: number;
  }


  interface IOverviewTabRegional {
    /**
    description: WHAT PERCENTAGE OF PEOPLE IN WAKISO LIVE BELOW THE NATIONAL POVERTY LINE? 
can be no data or '12%'
  */
    poorestPeople: IIndicatorValueWithToolTip;
    /**
    description: WHAT RESOURCES ARE AVAILABLE TO LOCAL GOVERNMENTS IN WAKISO? eg 3.6m or 2.7bn
this is a total of local, donor and central government resources
  */
    regionalResources: IIndicatorValueNCUWithToolTip;
    /**
    description: IndicatorDataColored is defined in country profile types
has local government, donor and central government
  */
    regionalResourcesBreakdown: Array<IIndicatorDataWithToolTip>;
    /**
    description: HOW MUCH DOES THE LOCAL GOVERNMENT SPEND PER PERSON?
  */
    localGovernmentSpendPerPerson: IIndicatorValueWithToolTip;
  }


  interface IIndicatorValueWithToolTip {
    value: string;
    toolTip: IToolTip;
  }


  interface IToolTip {
    source: string;
    heading: string;
  }


  interface IIndicatorValueNCUWithToolTip {
    value: string;
    value_ncu: string;
    toolTip: IToolTip;
  }


  interface IIndicatorDataWithToolTip {
    data: Array<IIndicatorData>;
    toolTip: IToolTip;
  }


  interface IIndicatorData {
    year: number;
    value: number;
    id: string;
    name: string;
    color: string | null;
    uid: string;
  }


  interface IDomestic {
    /**
    description: eg Actual or budget
  */
    budget_type: string;
    levels: Array<string>;
    year: number;
    value: number;
    value_ncu: number;
    uid: string;
    color: string;
  }


  interface IPopulationDistribution {
    group: string;
    value: number;
    year: number;
  }


  interface IPopulationDistributionWithToolTip {
    data: Array<IPopulationDistribution>;
    toolTip: IToolTip;
  }


  interface IPovertyTabKe {
    poorestPeople: IIndicatorValueWithToolTip;
    /**
    description: WHAT IS THE AVERAGE LIFE EXPECTANCY? 
  */
    meanExpenditure: IIndicatorValueWithToolTip;
    /**
    description: WHAT IS THE STANDARD OF LIVING SCORE? 
  */
    povertyGap: IIndicatorValueWithToolTip;
  }


  interface IPovertyTabUg {
    poorestPeople: IIndicatorValueWithToolTip;
    /**
    description: WHAT IS THE AVERAGE LIFE EXPECTANCY? 
  */
    lifeExpectancy: IIndicatorValueWithToolTip;
    /**
    description: WHAT IS THE STANDARD OF LIVING SCORE? 
  */
    stdOfLiving: IIndicatorValueWithToolTip;
  }


  interface IPopulationTabRegionalKe {
    /**
    description: The total population of a given district and the population density in per sq km
  */
    totalPopulation: IIndicatorValueWithToolTip;
    populationDensity: IIndicatorValueWithToolTip;
    populationBirthRate: IIndicatorValueWithToolTip;
  }


  interface IPopulationTabRegionalUg {
    /**
    description: The total population of a given district and the population density in per sq km
  */
    totalPopulation: IIndicatorValueWithToolTip;
    populationDensity: IIndicatorValueWithToolTip;
    /**
    description: Urban vs Rural population level
  */
    populationDistribution: IPopulationDistributionWithToolTip;
    averageDependencyRatio: IIndicatorValueWithToolTip;
    allAverageDependencyRatio: IIndicatorValueWithToolTip;
  }


  interface IEducationTabRegionalUg {
    /**
    description: WHAT IS THE PUPIL–TEACHER RATIO IN PRIMARY EDUCATION?...in government schools  and...in all schools 
  */
    pupilTeacherRatioGovtSchl: IIndicatorValueWithToolTip;
    pupilTeacherRatioOtherSchl: IIndicatorValueWithToolTip;
    /**
    description: WHAT PERCENTAGE OF STUDENTS PASS THE PRIMARY LEAVING EXAM? 
  */
    studentsPassRate: IIndicatorValueWithToolTip;
    studentsPassDistrictRank: IIndicatorValueWithToolTip;
    /**
    description: HOW MUCH PRIMARY EDUCATION FUNDING IS THERE? 
  */
    primaryEducationfunding: IIndicatorValueNCUWithToolTip;
  }


  interface IEducationTabRegionalKe {
    /**
    description: WHAT IS THE PUPIL–TEACHER RATIO IN PRIMARY EDUCATION?...in government schools  and...in all schools 
  */
    primaryPupilTeacherRatioAllSchl: IIndicatorValueWithToolTip;
    primaryTeacherRatioPublicSchl: IIndicatorValueWithToolTip;
    primaryTeacherRatioPrivateSchl: IIndicatorValueWithToolTip;
  }


  interface IHealthTabRegionalUg {
    /**
    description: WHAT IS THE DISTRICT LEAGUE HEALTH PERFORMANCE SCORE?
  */
    districtPerformance: IIndicatorValueWithToolTip;
    /**
    description: WHAT PERCENTAGE OF TUBERCULOSIS CASES HAVE BEEN SUCCESSFULLY TREATED? 
  */
    treatmeantOfTb: IIndicatorValueWithToolTip;
    districtHealthRank: IIndicatorValueWithToolTip;
    /**
    description: HOW MUCH LOCAL GOVERNMENT HEALTHCARE FUNDING IS THERE? 
  */
    healthCareFunding: IIndicatorValueNCUWithToolTip;
  }


  interface IHealthTabRegionalKe {
    birthAttendanceSkilled: IIndicatorValueWithToolTip;
    contraceptiveUse: IIndicatorValueWithToolTip;
    healthCareFunding: IIndicatorValueNCUWithToolTip;
  }


  interface ILocalGovernmentFinance {
    startYear: number;
    currencyUSD: string;
    currencyCode: string;
    /**
    description: come from finance file
  */
    revenueAndGrants: Array<IDomestic>;
    expenditure: Array<IDomestic>;
  }
}

// tslint:enable
\\""
`;

exports[`gql Types to Typescript types should return merged typedefs 1`] = `
"\\"schema {
  query: Query
}

type Query {
  dvptCooperation(country: String!): [BasicIndicator]
  topTchRecipients(country: String!): [TopTchRecipients]
  # id is district slug
  overviewTabRegional(id: String!, country: String!): OverviewTabRegional
}

type BasicIndicator {
  id: String!
  year: Int!
  color: String!
  value: Float!
  uid: String!
}

type TopTchRecipients {
  recipient: String!
  value: Float!
}

type IndicatorValueNCUWithToolTip {
  value: String!
  value_ncu: String!
  toolTip: ToolTip!
}

type ToolTip {
  source: String!
  heading: String!
}

type IndicatorData {
  year: Int!
  value: Float!
  id: String!
  name: String!
  color: String
  uid: String!
}

type Domestic {
  # eg Actual or budget
  budget_type: String!
  levels: [String]!
  year: Int!
  value: Float!
  value_ncu: Float!
  uid: String!
  color: String!
}

type PopulationDistribution {
  group: String!
  value: Float!
  year: Int!
}

type PopulationDistributionWithToolTip {
  data: [PopulationDistribution]!
  toolTip: ToolTip!
}

type IndicatorDataWithToolTip {
  data: [IndicatorData]!
  toolTip: ToolTip!
}

type IndicatorValueWithToolTip {
  value: String!
  toolTip: ToolTip!
}

type OverviewTabRegional {
  # WHAT PERCENTAGE OF PEOPLE IN WAKISO LIVE BELOW THE NATIONAL POVERTY LINE? 
   # can be no data or '12%'
  poorestPeople: IndicatorValueWithToolTip!
  # WHAT RESOURCES ARE AVAILABLE TO LOCAL GOVERNMENTS IN WAKISO? eg 3.6m or 2.7bn
   # this is a total of local, donor and central government resources
  regionalResources: IndicatorValueNCUWithToolTip!
  # IndicatorDataColored is defined in country profile types
   # has local government, donor and central government
  regionalResourcesBreakdown: [IndicatorDataWithToolTip]!
  # HOW MUCH DOES THE LOCAL GOVERNMENT SPEND PER PERSON?
  localGovernmentSpendPerPerson: IndicatorValueWithToolTip!
}

type PovertyTabKe {
  poorestPeople: IndicatorValueWithToolTip!
  # WHAT IS THE AVERAGE LIFE EXPECTANCY? 
  meanExpenditure: IndicatorValueWithToolTip!
  # WHAT IS THE STANDARD OF LIVING SCORE? 
  povertyGap: IndicatorValueWithToolTip!
}

type PovertyTabUg {
  poorestPeople: IndicatorValueWithToolTip!
  # WHAT IS THE AVERAGE LIFE EXPECTANCY? 
  lifeExpectancy: IndicatorValueWithToolTip!
  # WHAT IS THE STANDARD OF LIVING SCORE? 
  stdOfLiving: IndicatorValueWithToolTip!
}

type PopulationTabRegionalKe {
  # The total population of a given district and the population density in per sq km
  totalPopulation: IndicatorValueWithToolTip!
  populationDensity: IndicatorValueWithToolTip!
  populationBirthRate: IndicatorValueWithToolTip!
}

type PopulationTabRegionalUg {
  # The total population of a given district and the population density in per sq km
  totalPopulation: IndicatorValueWithToolTip!
  populationDensity: IndicatorValueWithToolTip!
  # Urban vs Rural population level
  populationDistribution: PopulationDistributionWithToolTip!
  averageDependencyRatio: IndicatorValueWithToolTip!
  allAverageDependencyRatio: IndicatorValueWithToolTip!
}

type EducationTabRegionalUg {
  # WHAT IS THE PUPIL–TEACHER RATIO IN PRIMARY EDUCATION?...in government schools  and...in all schools 
  pupilTeacherRatioGovtSchl: IndicatorValueWithToolTip!
  pupilTeacherRatioOtherSchl: IndicatorValueWithToolTip!
  # WHAT PERCENTAGE OF STUDENTS PASS THE PRIMARY LEAVING EXAM? 
  studentsPassRate: IndicatorValueWithToolTip!
  studentsPassDistrictRank: IndicatorValueWithToolTip!
  # HOW MUCH PRIMARY EDUCATION FUNDING IS THERE? 
  primaryEducationfunding: IndicatorValueNCUWithToolTip!
}

type EducationTabRegionalKe {
  # WHAT IS THE PUPIL–TEACHER RATIO IN PRIMARY EDUCATION?...in government schools  and...in all schools 
  primaryPupilTeacherRatioAllSchl: IndicatorValueWithToolTip!
  primaryTeacherRatioPublicSchl: IndicatorValueWithToolTip!
  primaryTeacherRatioPrivateSchl: IndicatorValueWithToolTip!
}

type HealthTabRegionalUg {
  # WHAT IS THE DISTRICT LEAGUE HEALTH PERFORMANCE SCORE?
  districtPerformance: IndicatorValueWithToolTip!
  # WHAT PERCENTAGE OF TUBERCULOSIS CASES HAVE BEEN SUCCESSFULLY TREATED? 
  treatmeantOfTb: IndicatorValueWithToolTip!
  districtHealthRank: IndicatorValueWithToolTip!
  # HOW MUCH LOCAL GOVERNMENT HEALTHCARE FUNDING IS THERE? 
  healthCareFunding: IndicatorValueNCUWithToolTip!
}

type HealthTabRegionalKe {
  birthAttendanceSkilled: IndicatorValueWithToolTip!
  contraceptiveUse: IndicatorValueWithToolTip!
  healthCareFunding: IndicatorValueNCUWithToolTip!
}

type LocalGovernmentFinance {
  startYear: Int!
  currencyUSD: String!
  currencyCode: String!
  # come from finance file
  revenueAndGrants: [Domestic]!
  expenditure: [Domestic]!
}
\\""
`;
